src/coreclr/jit/lsrabuild.cpp

```diff
@@ -1456,6 +1456,34 @@ Interval* LinearScan::getUpperVectorInterval(unsigned varIndex)
     unreached();
 }
 
+//------------------------------------------------------------------------
+// shouldBuildUpperVectorSaveRestore - Check if vector save/restore is needed.
+//
+// Arguments:
+//    tree       - The current node being handled
+//
+// Notes: Vector save/restore is usually not needed before the call that never returns.
+//        It is also not needed if the block ends with a throw.
+bool LinearScan::shouldBuildUpperVectorSaveRestore(GenTree* tree)
+{
+    if ((tree != nullptr) && tree->IsCall())
+    {
+        if (tree->AsCall()->IsNoReturn())
```

Comment:
You might also want to check `fgIsThrow` here but I assume both `IsNoreReturn` and `fgIsThrow` should always be inside a `BBJ_THROW` block

Reply:
I looked around and seems that the only non-throw case where calls are marked as NoReturn is here:
https://github.com/dotnet/runtime/blob/24ce527fb5f966a8262e48b5812cefc240ba6ef5/src/coreclr/jit/fgbasic.cpp#L2138

I am not familiar with how that translates to during LSRA.